File: tournaments/countDistantPairs/countDistantPairs.js
function countDistantPairs(inputString, distance) {
    let result = 0;

    for (let i = 0; inputString.length > i + distance + 1; i++) {
        if (inputString.substring(i, i + 1) === inputString.substring(i + distance + 1, i + distance + 2)) {
            ++result;
        }
    }

    return result;
}

File: tournaments/countDistantPairs/countDistantPairs.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub countDistantPairs {
    my ($inputString, $distance) = @_;

    my $result = 0;

    for (my $i = 0; length($inputString) > $i + $distance + 1; ++$i) {
        if (substr($inputString, $i, 1) eq substr($inputString, $i + $distance + 1, 1)) {
            ++$result;
        }
    }

    return $result;
}
